core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -588,6 +600,7 @@ private void timeoutTask(TaskDef taskDef, Task task) {
                 .withTaskToSchedule(taskToSchedule)
                 .withTaskInput(input)
                 .withRetryCount(retryCount)
+                .withIterationCount(iterationCount)
                 .withRetryTaskId(retriedTaskId)
                 .withTaskId(taskId)
                 .withDeciderService(this)
```

Comment:
There is a following statement at the end of method:
```
        List<Task> tasks = taskMappers.get(taskType.name()).getMappedTasks(taskMapperContext).stream()
                .filter(task -> !tasksInWorkflow.contains(task.getReferenceTaskName()))
                .collect(Collectors.toList());
```
Task with referenceTaskName _LeReview_ is already completed. Goto task again schedule it based on `gotoTask` field value. However, due to above filter it does not get scheduled.

Am I missing something?

Reply:
@sharmasourabh Thanks for pointing it out. Actually this method has been changed recently, previously the filter was based on inProgress tasks only. So, if a task is not in progress, it can be scheduled again. I missed this change during the merge. I will update the PR with the change.